hibernate-core/src/main/java/org/hibernate/type/format/AbstractJsonFormatMapper.java

```diff
@@ -27,7 +27,7 @@ public final <T> T fromString(CharSequence charSequence, JavaType<T> javaType, W
 	@Override
 	public final <T> String toString(T value, JavaType<T> javaType, WrapperOptions wrapperOptions) {
 		final Type type = javaType.getJavaType();
-		if ( type == String.class || type == Object.class ) {
+		if ( value instanceof String ) {
```

Comment:
For the sake of this bug-fix, let's use this please.

```suggestion
		if ( type == String.class ) {
```